org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/config/ZclReportingConfig.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.internal.converter.config;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+
+/**
+ * Configuration handler for reporting of digital attributes. This should be applied for each channel - ie if a channel
+ * uses multiple attributes, then they should all be configured the same.
+ *
+ * @author Chris Jackson
+ *
+ */
+public class ZclReportingConfig implements ZclClusterConfigHandler {
+    private final Logger logger = LoggerFactory.getLogger(ZclReportingConfig.class);
+
+    private static final Integer REPORTING_RANGE_MIN = 1;
+    private static final Integer REPORTING_RANGE_MAX = 86400;
+
+    private static final Integer REPORTING_DEFAULT_MIN = 1;
+    private static final Integer REPORTING_DEFAULT_MAX = 900;
+
+    private static final Integer POLLING_RANGE_MIN = 15;
+    private static final Integer POLLING_RANGE_MAX = 86400;
+    private static final Integer POLLING_DEFAULT = 900;
+
+    private static final String CONFIG_ID = "zigbee_reporting_";
+    private static final String CONFIG_REPORTINGMIN = CONFIG_ID + "min";
+    private static final String CONFIG_REPORTINGMAX = CONFIG_ID + "max";
+    private static final String CONFIG_REPORTINGCHANGE = CONFIG_ID + "change";
+    public static final String CONFIG_POLLING = CONFIG_ID + "polling";
+
+    private int reportingTimeMin = REPORTING_DEFAULT_MIN;
+    private int reportingTimeMax = REPORTING_DEFAULT_MAX;
+    private int reportingChange = 1;
+    private int pollingPeriod = POLLING_DEFAULT;
+
+    // Reporting change configuration. These will need to be set for each channel.
+    private boolean isAnalogue = false;
+    private BigDecimal defChange;
```

Comment:
Small suggestion: Call variable `defaultChange`, to make a little more clear what the variable is for.

Reply:
I could change I guess - to me it was quite clear, and in line with the reduced names on minimum and maximum to min and max. Do you really think this name needs to be extended?